### PR TITLE
Initial plumbing for face-elem alg/kernels.

### PR DESCRIPTION
**Pull request by alanw0**
add MomentumSymmetryElemKernel to LowMachEquationSystem.C

also add line command: use_consolidated_face_elem_bc_algorithm: yes/no
which turns on AssembleFaceElemSolverAlgorithm and MomentumSymmetry
kernel in LowMachEquationSystem.C.

rebless diffs in elemHybridFluids, elemHybridFluidsShift and
dgNonConformalThreeBlade. These are the tests which activate
consolidated
solver algorithms and also use symmetry_boundary_condition.
In this scenario, the face-sorting code was being activated but now
is not activated unless the new (above-mentioned) line command is
added to the input file.

### CODE REVIEW

File: include/utils/StkHelpers.h [hunk 1]
```diff
@@ -29,6 +34,28 @@ void populate_ghost_comm_procs(const stk::mesh::BulkData& bulk_data, stk::mesh::
...
+
+  stk::topology elemTopo = blockParts[0]->topology();
+  if (blockParts.size() > 1) {
+    for(size_t i=1; i<blockParts.size(); ++i) {
+      ThrowRequireMsg(blockParts[i]->topology() == elemTopo,
+                 "Error, found blocks of different topology connected to surface '"
+                  <<surfacePart.name()<<"', "<<elemTopo<<" and "<<blockParts[i]->topology());
+    }
+  }
+  //std::cerr<<"for surfacePart "<<surfacePart.name()<<", got topo "<<blockParts[0]->topology()<<std::endl;
```
Comment (spdomin) [inline include/utils/StkHelpers.h]: Please remove comment.

File: include/kernel/KernelBuilder.h [hunk 2]
```diff
@@ -88,6 +89,32 @@ namespace nalu{
...
   }
 
+  template <template <typename> class T, typename... Args>
+  Kernel* build_face_elem_topo_kernel(int dimension,
+                                      stk::topology faceTopo, stk::topology elemTopo,
+                                      Args&&... args)
+  {
+    switch(faceTopo.value()) {
+      case stk::topology::QUAD_4:
+        return new T<AlgTraitsQuad4Hex8>(std::forward<Args>(args)...);
```
Comment (spdomin) [inline include/kernel/KernelBuilder.h]: We will need to think about this for the general face/elem approach as a quad4 can be attached to a wedge or pyramid. @rcknaus might have views here.

**Comment by spdomin** [comment]
@alanw0 - actually, I do not see the input file changes to add the new consolidatedBc option?

**Comment by alanw0** [comment]
@spdomin thanks for the comments, I have added two commits to address them.

**Pull request merged**